src/pocketmine/block/Button.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+namespace pocketmine\block;
+
+use pocketmine\item\Item;
+use pocketmine\item\Tool;
+use pocketmine\level\Level;
+use pocketmine\math\Vector3;
+use pocketmine\Player;
+
+class Button extends Flowable{
+
+	protected $id = self::STONE_BUTTON;
+
+	public function __construct($meta = 0){
+		$this->meta = $meta;
+	}
+
+	public function getHardness(){
+		return 0.5;
+	}
+
+	public function getResistance(){
+		return 2.5;
+	}
+
+	public function canBeActivated(){ //TODO: Redstone
+		return false;
+	}
+
+	public function place(Item $item, Block $block, Block $target, $face, $fx, $fy, $fz, Player $player = null){
+		$below = $this->getSide(Vector3::SIDE_DOWN);
+		if (!$target->isTransparent() || ($face === Vector3::SIDE_DOWN && ($below instanceof WoodenSlab && ($below->meta & 0x08) === 0x08) || ($below instanceof Stair && ($below->meta & 0x04) === 0x04) || $below instanceof Fence || $below instanceof CobblestoneWall)){
+			$this->meta = $face;
+			$this->getLevel()->setBlock($block, $this, true, false);
+			return true;
+		}
+
+		return false;
+	}
+
+	public function onUpdate($type){
+		if ($type === Level::BLOCK_UPDATE_NORMAL){
+			$below = $this->getSide(Vector3::SIDE_DOWN);
+			if ($this->getSide($this->meta)->isTransparent() && !($this->meta === 0 && ($below instanceof WoodenSlab && ($below->meta & 0x08) === 0x08) || ($below instanceof Stair && ($below->meta & 0x04) === 0x04) || $below instanceof Fence || $below instanceof CobblestoneWall)){
+				$this->getLevel()->useBreakOn($this);
+
+				return Level::BLOCK_UPDATE_NORMAL;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Returns the best tool type to use for breaking this type of block.
+	 * @return int
+	 */
+	public function getToolType(){
+		return $this->getId() === Block::STONE_BUTTON ? Tool::TYPE_PICKAXE : Tool::TYPE_AXE;
```

Comment:
It only makes sense to squash blocks into the same class type when they share all the same properties and no difference in code. This is hacky and reduces scalability.